### DISCUSSION TITLE
How to fix Bootstrap 5 Dropdown didnt work after wire:navigate page?

### DISCUSSION THREAD

**Discussion by ilahazs** [livewire/livewire#7513] [1. Help] [answered]
https://github.com/livewire/livewire/assets/68785569/1bab9347-f5d7-4f59-a5e9-ef822cfc5347

I've bootstrap 5 navbar and on the menu item there's dropdown element. It's work on the initial load. 
But after I navigate to other page and click the dropdown again. 
The dropdown stopped working. I need to navigate to another page, and click it again to make it work.

I'm using Livewire 3 and Laravel 10.

I've done some research that said I need to reinitialize Bootstrap after navigating with Livewire.

Like this :

`document.addEventListener('livewire:navigated', () => {
   console.log('Livewire navigated.');
   initDropdown();
});

function initDropdown() {
   try {
      let myDropdown = document.querySelector('#dropdown-indeks');
      new bootstrap.Dropdown(myDropdown);
      console.log('initDropdown');
   } catch (error) {
      console.error('Theres some error:', error);
   }
}`

But unfortunately it didnt work at all. Please help me guys

**Comment by snoedev** [comment, marked as answer]
Add `data-navigate-once` to your Bootstrap JS. It works for me

`<script data-navigate-once src="bootstrap.js"></script>`

**Reply by techghor** [reply]
not working for me..

**Reply by unifydevelopment** [reply]
I stuck on this issue for more than 3 days. Thanks buddy. it works for me 👍

**Reply by rakaarifbudiman** [reply]
Thank you very much, you are truly very helpful, it worked.

**Reply by abdulrehman25** [reply]
Thank you @snoeDev , it worked for me as well Livewire 3

**Reply by PhillipMwaniki** [reply]
This is going to be a hard one if your script comes from the app.js file. Does anyone know how to tackle it from there ?

**Reply by allphptricks** [reply]
Your solution worked very well, thanks

**Reply by antoniuspj** [reply]
just faced the same issue, and it works. thank you bro!

**Reply by mbilalshafiq9** [reply]
Thanks, It worked

**Reply by Thirumalai15** [reply]
Thanks, It worked for me!! 
I am using tabler UI dashboard

**Reply by harshad-pindoriya** [reply]
Thanks, man. It worked for me as well.

**Reply by filatofanton** [reply]
Thanks a lot!

**Reply by Jasen-Angelov** [reply]
You have no idea how many hours I spent in despair because of this issue! Thank you!!!

**Reply by arjunstein** [reply]
nice answer thanks

**Reply by repulse-86** [reply]
main this is the answer. thank you so much

**Reply by GunaIJ** [reply]
Great!

**Reply by PRINCE-CODER1** [reply]
ohhhh Damn Man Thanks bruh

**Reply by lehuylxag1995** [reply]
Thanhks pro.... i working 👍

**Reply by ariwahidin** [reply]
thank broo

**Reply by rajafakhri** [reply]
Thank Bro

**Reply by dreigningking** [reply]
This just saved me

**Reply by Aryo07** [reply]
data-navigate-once="true"

thanks bro... i working 👍

**Comment by edulazaro** [comment]
You can wrap the heading using the @persist tag, like:

```html
@persist('nav')
<!-- Your nav code here -->
@endpersist
```

This way, it will only be rendered once across all renders when using `wire:navigate`.

**Comment by unifydevelopment** [comment]
You can wrap the heading using the @persist tag, like:

@persist('nav')

@endpersist
